ext/sodium/libsodium.c

```diff
@@ -0,0 +1,2682 @@
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include "php.h"
+#include "php_ini.h"
+#include "ext/standard/info.h"
+#include "php_libsodium.h"
+#include "zend_exceptions.h"
+
+#include <sodium.h>
+#include <stdint.h>
+
+#define ZSTR_TRUNCATE(zs, len) do { ZSTR_LEN(zs) = (len); } while(0)
```

Comment:
Having that moved to other Zend macros would make sense, as it can be useful to other extensions.